src/auth.rs

```diff
@@ -67,30 +67,42 @@ pub enum AuthError {
 impl From<AuthError> for MomentoError {
     fn from(err: AuthError) -> Self {
         match err {
-            AuthError::LoginAborted => MomentoError::Cancelled {
-                description: "aborted login".into(),
-                source: err.into(),
-            },
-            AuthError::LoginFailed(_) => MomentoError::PermissionDenied {
-                description: "login failed".into(),
-                source: err.into(),
-            },
-            AuthError::BadUri(_) => MomentoError::BadRequest {
-                description: "bad uri".into(),
-                source: Some(err.into()),
-            },
-            AuthError::Connection(_) => MomentoError::InternalServerError {
-                description: "connection failed".into(),
-                source: err.into(),
-            },
-            AuthError::ServerError(_) => MomentoError::InternalServerError {
-                description: "server error".into(),
-                source: err.into(),
-            },
-            AuthError::ActionError(_) => MomentoError::ClientSdkError {
-                description: "login action failed".into(),
-                source: err.into(),
-            },
+            AuthError::LoginAborted => MomentoError::Cancelled(SdkError {
+                message: "aborted login".into(),
+                error_code: MomentoErrorCode::CancelledError,
+                inner_exception: Some(err.into()),
```

Comment:
supernit but in rust there really isn't such a thing as an exception, so let's call this field `inner_error` maybe?